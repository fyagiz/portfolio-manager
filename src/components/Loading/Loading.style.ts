import { StyleSheet } from "react-native";
import { COLOR, STYLE } from "../../utils/constants";

export default StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: COLOR.loadingBackgroundColor,
    opacity: STYLE.opacity,
  },
});
