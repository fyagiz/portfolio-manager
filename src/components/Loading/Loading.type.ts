export type LoadingsPropsType = {
  isLoading: boolean;
};
