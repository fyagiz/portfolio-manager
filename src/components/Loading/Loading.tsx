import { ActivityIndicator, Modal, View } from "react-native";
import STYLE from "./Loading.style";
import { LoadingsPropsType } from "./Loading.type";

const Loading = (props: LoadingsPropsType) => {
  const { isLoading } = props;

  return (
    <Modal visible={isLoading} transparent>
      <View style={STYLE.container}>
        <ActivityIndicator size={"large"} />
      </View>
    </Modal>
  );
};

export default Loading;
