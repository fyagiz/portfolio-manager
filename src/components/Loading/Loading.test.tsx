import { render } from "@testing-library/react-native";
import Loading from "./Loading";

/*
1- Loading should be rendered
*/

describe("Loading Component Unit Tests", () => {
  test("should render correctly", () => {
    const loadingComponent = render(<Loading isLoading />);
    expect(loadingComponent).toMatchSnapshot();
  });
});
