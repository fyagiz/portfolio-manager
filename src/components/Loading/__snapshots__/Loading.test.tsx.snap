// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component Unit Tests should render correctly 1`] = `
<Modal
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <ActivityIndicator
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  />
</Modal>
`;
